### DISCUSSION TITLE
World map

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#101] [Q&A] [answered]
Routing from locations outside the active vector map range are rejected.  
With a new Cruiser install, wouldn't it be optimal to install_ activate the Mapsforge World map by default ?

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the suggestion.

Current [world.map](https://download.mapsforge.org/maps/v5/world/) is basic with sea and land.

**Reply by devemux86** [reply]
We can allow imports outside of the loaded offline map area and possibly display a message to load an offline map for the area.

The current world map was an experiment, it needs improvements before it can be used.

**Reply by MotoUKRider** [reply]
For info+comparison.... The Global Map in Garmin's BaseCamp offers reasonably resolution, certainly as far as size+border goes, but roads+detail are low-res.

Overview:
![garmin-world-map-1](https://github.com/devemux86/cruiser/assets/125696109/beef219b-3cae-4749-8a17-44032d4cfddd)

London, detail:
![garmin-world-map-2](https://github.com/devemux86/cruiser/assets/125696109/e477a0ad-a196-4a4f-b1cb-55da16bb4665)

Calais, with route:
![garmin-world-map-3](https://github.com/devemux86/cruiser/assets/125696109/609afc70-32ec-4acb-bcee-2e4593da0226)

**Comment by ToToGSARider** [comment]
Verstehe ich das richtig, dass man dann nur noch die Karten und nicht mehr die POIS aktualisieren muss?

Do I understand correctly that you then only have to update the maps and no longer the POIS?

**Reply by devemux86** [reply]
They are separate formats and have a different purpose:
- Map files: offline map rendering
- POI files: offline search

I meant that I wrote a new format for POI files, which will help make searching faster and easier to use.

---

Es handelt sich um getrennte Formate, die einen unterschiedlichen Zweck haben:
- Kartendateien: Offline-Kartendarstellung
- POI-Dateien: Offline-Suche

Ich meinte damit, dass ich ein neues Format für POI-Dateien geschrieben habe, das die Suche schneller und einfacher machen wird.

**Comment by devemux86** [comment]
World map is available in the download manager in the latest:
- https://github.com/devemux86/cruiser/discussions/390

**Comment by devemux86** [comment]
I asked [OpenAndroMaps](https://www.openandromaps.org/en/) and [Freizeitkarte](https://www.freizeitkarte-osm.de/android/en/index.html) if they want to host the world map on their servers too.

It will then automatically become available to everyone.

**Reply by kds112** [reply]
Wouldn't it be possible to activate the mapsforge-World together with other OAM-maps?

**Reply by devemux86** [reply]
@kds112 it is complicated to mix offline maps from different providers.

I already contacted OpenAndroMaps and Freizeitkarte to host the world map in their servers.
Then Cruiser (and other apps) can see it automatically and download it.

**Reply by kds112** [reply]
Okay... Hopefully OAM and Freizeitkarte will accept the hosting👍

**Comment by ramondevic** [comment]
I manually downloaded World.map from mapsforge.org but it only shows the coastline, nothing else.
It would be good if you could at least see the border limits and the name of the countries, would that be possible?

I have opened the World.map and the one of Andorra.

![Screenshot_20231205-172334](https://github.com/devemux86/cruiser/assets/122691401/70f0b7aa-af26-4e65-bb0b-6df09b04f814)

**Reply by devemux86** [reply]
@ramondevic indeed countries, borders, capitals are already planned.

**Reply by ramondevic** [reply]
It will be perfect!

**Comment by kds112** [comment]
It is really great that you can combine now the world with OAM maps 👍👍👍
Because it is a "hidden" feature I propose to add a selection "Show world background" to the map settings. If selected you are asked whether the map should be downloaded and activated. If you unselect the map is deactivated and you are asked whether the map should be deleted.

**Reply by kds112** [reply]
And an update of the world map with some more content would be great 🙃😉😆

**Comment by devemux86** [comment]
There is also the thought after I update it with more content and if its size remains reasonable
to be included in the app and activated on new installations instead of the current online map.

**Reply by kds112** [reply]
An even better solution👍

**Reply by kds112** [reply]
But what about older devices with limited memory? Maybe the way with an selectable downloadable World map should be preferred...

**Reply by kds112** [reply]
Maybe you can offer one empty World map (as is) and another one with some content (borders, capitals, large rivers) for better orientation.

**Comment by devemux86** [comment, marked as answer]
World map is available in the download manager of [Cruiser 3.0.37](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by kds112** [reply]
Will you expand this nice feature to a a map setting? In the moment it is quite hidden just for Cruiser experts...

**Reply by devemux86** [reply]
> Will you expand this nice feature to a a map setting?

It could be done, we also need to create a world map with more content.

**Comment by kds112** [comment]
As 0709wiwiwi proposes in #426, please integrate the world map directly in the app (and remove it from the download folders). And future updates of the world map could be done with an app update. 🌎🌍🌏

**Comment by kds112** [comment]
Any plans to integrate it in the next version? 🙃

**Reply by devemux86** [reply]
@kds112 we will see, many new features are coming! :slightly_smiling_face: 

I also need to try some things with the world map first.

**Comment by kds112** [comment]
Another idea. In the world map you can see all downloaded and activated maps. But could you also display the regions of downloaded but deactivated maps? Maybe as a (gray) shaded area? Then you can recognize all areas with available (=downloaded) map Information. What do you think? 😀

**Reply by devemux86** [reply]
@kds112 thanks, this is an interesting idea.

Something similar exists in the BRouter download map (with tiles) and these color codes:
- https://github.com/devemux86/cruiser/discussions/137#discussioncomment-6029728

**Comment by kds112** [comment]
Any implemetation plans for the next release...?  😀

**Reply by devemux86** [reply]
@kds112 I must first see how to create a more detailed world map.

Then we'll see how it can be integrated better in Cruiser. It has work. :slightly_smiling_face:

**Comment by kds112** [comment]
Any chance for the upcoming version 3.0.42? 😉

**Reply by devemux86** [reply]
@kds112 when a feature is ready, it will be announced. :slightly_smiling_face:

World map is already available, so features that are not yet available take priority.

**Reply by kds112** [reply]
Well, showing downloaded but deactivated maps as shaded areas would be a new feature... 😎

**Comment by devemux86** [comment]
[paws.cz](https://osm.paws.cz/) provides a Europe admin map with more details.

**Reply by bios4** [reply]
Can they be somehow "combined" with the e.g. Mapsforge offline maps @devemux86?

The map shows country borders and (the biggest) cities, but no details, roads, anything when zooming in at my current location.
Which makes them practically useless for navigation.

Or did I miss something I have to set/activate in addition when using such external *.map files?

**Reply by devemux86** [reply]
You can either use the downloaded maps or open external maps.

Someone needs to create a new more detailed world map and we can host it at Mapsforge server.

**Reply by bios4** [reply]
So this means the linked Europe map is really meant to give just a brief overview!?
It is OK that I don't see anything (except hillshading) at higher zoom levels?

**Reply by devemux86** [reply]
It can be used if you open multiple external map files.
